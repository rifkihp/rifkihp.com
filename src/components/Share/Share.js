// @flow
import { graphql, StaticQuery } from 'gatsby';
import React from 'react';
import {
  FacebookShareButton,
  LinkedinShareButton,
  RedditShareButton,
  TwitterShareButton,
} from 'react-share';

import { logEvent } from '../../utils/log';
import styles from './Share.module.scss';

type Props = {
  +title: string,
  +url: string,
  +shareText?: string,
};

type PureProps = {
  +twitter: string,
} & Props;

const PureShare = ({ title, url, shareText, twitter }: PureProps) => (
  <div className={styles['share']}>
    <h4 className={styles['share-title']}>{shareText || 'SHARE THIS POST'}</h4>
    <div className={styles['share-buttons']}>
      <FacebookShareButton
        beforeOnClick={() => logEvent('ShareButton', 'facebook-click')}
        className={styles['facebook']}
        resetButtonStyle={false}
        url={url}
      >
        Facebook
      </FacebookShareButton>
      <TwitterShareButton
        beforeOnClick={() => logEvent('ShareButton', 'twitter-click')}
        className={styles['twitter']}
        resetButtonStyle={false}
        url={url}
        title={title}
        via={twitter}
      >
        Twitter
      </TwitterShareButton>
      <LinkedinShareButton
        beforeOnClick={() => logEvent('ShareButton', 'linkedin-click')}
        className={styles['linkedin']}
        resetButtonStyle={false}
        url={url}
        title={title}
      >
        LinkedIn
      </LinkedinShareButton>
      <RedditShareButton
        beforeOnClick={() => logEvent('ShareButton', 'reddit-click')}
        className={styles['reddit']}
        resetButtonStyle={false}
        url={url}
        title={title}
      >
        Reddit
      </RedditShareButton>
    </div>
  </div>
);

export const Share = ({ url, ...props }: Props) => (
  <StaticQuery
    query={graphql`
      query ShareQuery {
        site {
          siteMetadata {
            url
            author {
              contacts {
                twitter
              }
            }
          }
        }
      }
    `}
    render={data => (
      <PureShare
        {...props}
        url={data.site.siteMetadata.url + url}
        twitter={data.site.siteMetadata.author.contacts.twitter}
      />
    )}
  />
);

export default Share;
