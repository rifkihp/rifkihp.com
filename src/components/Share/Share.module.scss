@import '../../assets/scss/variables';
@import '../../assets/scss/mixins';

.share {
  @include margin-top(1);

  &-title {
    margin: 0;
    font-size: $typographic-base-font-size * 1;
    font-weight: bold;
    text-align: center;
  }

  &-buttons {
    display: flex;
    flex-direction: row;
    justify-content: center;

    > * {
      flex-basis: 25%;
      flex-grow: 1;
    }

    button {
      color: white; // don't invert
      width: 100%;
    }

    button.facebook {
      background-color: $color-facebook;

      &:hover {
        background-color: darken($color-facebook, 10%);
      }
    }

    button.twitter {
      background-color: $color-twitter;

      &:hover {
        background-color: darken($color-twitter, 10%);
      }
    }

    button.linkedin {
      background-color: $color-linkedin;

      &:hover {
        background-color: darken($color-linkedin, 10%);
      }
    }

    button.reddit {
      background-color: $color-reddit;

      &:hover {
        background-color: darken($color-reddit, 10%);
      }
    }
  }
}

@include breakpoint-sm {
  .share {
    @include margin-top(1.5);
  }
}
