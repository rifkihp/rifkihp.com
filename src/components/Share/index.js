// @flow
export { default } from './Share';
