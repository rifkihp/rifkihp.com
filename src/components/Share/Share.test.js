import React from 'react';
import renderer from 'react-test-renderer';

import { Share } from './Share';

describe('Share', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Share url="/test" title="Test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
