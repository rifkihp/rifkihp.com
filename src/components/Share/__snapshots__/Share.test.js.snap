// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share renders correctly 1`] = `
<StaticQuery
  render={[Function]}
/>
`;
